### ISSUE TITLE
Third party logins fail to preserve the referer URL

### ISSUE THREAD

**Issue by danieldegroot2** [openstreetmap/openstreetmap-website#4980]
Opening iD editor (or any other editor) now requires clicking Edit twice when a logged out user logs in with a third party.

**Actual behaviour**
To reproduce
1. Go to https://www.openstreetmap.org/
2. Click "Edit"
3. Log in (third party login)
4. Goes back to https://www.openstreetmap.org/
5. Click "Edit" again
6. Goes to https://www.openstreetmap.org/edit proper.

**Expected behaviour**
1. Go to https://www.openstreetmap.org/
2. Click "Edit"
3. Log in (third party login)
4. Goes to https://www.openstreetmap.org/edit straight away, as it used to.

It breaks the user flow, similar to https://github.com/openstreetmap/iD/issues/8994

Instance: openstreetmap.org
Browser: Chrome 126.0.6478.127
OS: Windows 10

Edit: Clarified the problem description.

cc: @tyrasd

**Comment by mmd-osm** [comment]
I've encountered a similar issue with https://community.openstreetmap.org/ and using third party log ins. After clicking on "Log in", the page redirects to https://www.openstreetmap.org (like step 3 + 4 above) instead of the community forum.

**Comment by tomhughes** [comment]
Please don't exaggerate problems when opening bugs - the original title here is completely misleading because it makes it sound like every launch of iD when it reality it's only when the user is not logged in.

It's also misleading because it makes it sound like it's about iD when it's really just generic to certain login methods not preserving the URL when redirecting after login.

Finally it's misleading because this is apparently only an issue with third party logins, though I realise the reporter may not have realised that, though they did specify that it should be a third party login.

**Comment by danieldegroot2** [comment]
Thank you for your reply. Sorry, I understand my report included some misleading text.

I personally can't properly troubleshoot because of my preferences. When authentication succeeded my initial impression was of a more generic issue. My initial title is specific due to my preference for iD and being able to test only iD directly.

I've edited also the first line of description to clarify it is about third party login and include the confirmation it includes other editors. Furthermore, I have removed remaining misleading text about unrelated changes to the website.
